example/config_parser.cpp

```diff
@@ -0,0 +1,102 @@
+
+// Copyright (c) 2023 Jiri Nytra (jiri.nytra at gmail dot com)
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <flux.hpp>
+
+#include <cassert>
+#include <map>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+#include <string_view>
+#include <variant>
+
+
+struct comment_t
+{
+    std::string content;
+};
+
+struct section_t
+{
+    std::string name;
+};
+
+struct option_t
+{
+    std::string key;
+    std::string value;
+};
+
+using token_t = std::variant<comment_t, section_t, option_t>;
+using config_t = std::map<std::string, std::string, std::less<>>;
+
+struct context_t
+{
+    std::string curr_section = "root";
+    config_t config;
+};
+
+template <class... Ts> struct overloaded : Ts...
+{
+    using Ts::operator()...;
+};
+
+template <class... Ts> overloaded(Ts...) -> overloaded<Ts...>;
+
+bool not_blank_line(const std::string& line)
+{
+    auto space = [](auto c) { return std::isspace(c); };
+    return not(line.empty() or flux::all(line, space));
+}
+
+token_t parse_line(const std::string& line)
+{
+    if (line.starts_with("#"))
+        return comment_t{line.substr(1)};
+    else if (line.starts_with("[") and line.ends_with("]"))
+        return section_t{line.substr(1, line.size() - 2)};
+
+    auto items = flux::from(line).split_string("=").to<std::vector>();
```

Comment:
It's a bit of a shame that we need to convert to a vector here rather than being able to use the sequence directly, but it does seem to be the tidiest way to do it and it also shows off the `to()` function so it's fine

Reply:
I agree it's not the most efficient way, but for demonstration purposes I'd say sufficient.

Reply:
Yep, definitely